src/__tests__/preset-by-number.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, expect, test } from '@jest/globals'
+import { parsePresetVariableOption } from '../preset-by-number.js'
+import { CompanionOptionValues } from '@companion-module/base'
+import { PresetRecall } from '../camera/commands.js'
+import { prettyBytes } from '../visca/utils.js'
+import { MockContext } from './mock-context.js'
+import type { MatcherFunction } from 'expect'
+import { Command } from '../visca/command.js'
+
+function optionsWithPresetOption(text: string): CompanionOptionValues {
+	return { val: text }
+}
+
+/**
+ * Custom jest matcher to check that options are not a string and that the command bytes match
+ * @param actual
+ * @param command
+ * @param expected_bytes
+ * @returns
+ */
+const toBeExpectedCommandBytes: MatcherFunction<[command: Command, expected_bytes: string]> = function (
+	actual,
+	command,
+	expected_bytes
+) {
+	if (typeof actual === 'string') {
+		throw new TypeError('Input showuld be of type CompanionOptionValues not string')
+	}
+
+	const commandBytes = command.toBytes(actual as CompanionOptionValues)
+	const pass = prettyBytes(commandBytes) === expected_bytes
+
+	return {
+		message: () =>
+			`expected ${this.utils.printReceived(prettyBytes(commandBytes))} ${
+				pass ? 'not ' : ''
+			}to match\nreceived ${this.utils.printExpected(expected_bytes)}`,
+		pass: pass,
```

Comment:
JavaScript/TypeScript allow you to just say "pass" as a shorthand to define a property with a name expression, with that name expression's value.

Reply:
done